sql/persisted_variable.cc

```diff
@@ -332,17 +334,19 @@ bool Persisted_variables_cache::set_variable(THD *thd, set_var *setvar) {
     uint dummy_err;
     String bool_str;
     if (setvar->value) {
-      res = setvar->value->val_str(&str);
+      setvar->var->persist_only_update(thd, setvar, str.ptr());
```

Comment:
There is a lot of code trying to get string representation. why the current code doesn't work? set_var->value->val_str(&str)?

Reply:
As I understand it value->val_str(&str) returns original value. When check method is invoked it stores resulting (and possibly changed) value in a different place (var->save_result) and it should be handled differently.

Reply:
Understood, so this about getting the string representation. Can you please extract out as separate PR/change?